OutOfSchool/OutOfSchool.BusinessLogic/Extensions/SearchResultExtensions.cs

```diff
@@ -4,5 +4,10 @@ namespace OutOfSchool.BusinessLogic.Extensions;
 public static class SearchResultExtensions
 {
     public static bool IsNullOrEntitiesEmpty<T>(this SearchResult<T> searchResult)
-        => searchResult == null || searchResult.Entities.Count == 0;
+        => IsNull(searchResult) || searchResult.Entities is null || searchResult.Entities.Count == 0;
+
+    public static bool IsNullOrTotalAmountIsZero<T>(this SearchResult<T> searchResult)
+        => IsNull(searchResult) || searchResult.TotalAmount == 0;
```

Comment:
(Logic) Please leave only one extension method checking for the `TotalAmount`. Subsequent changes in controller extensions should be done as well.

Explanations:

Providing 2 extension methods you (implicitly) specify that there could be such `SearchResult`, where `TotalAmount == 0` but `Entities` are not empty. 
I've checked all places where `SearchResult` is created and returned and it seems to be not the case.
Also, from semantic perspective, it should always `TotalAmount >= Entities.Count()`, otherwise the name `TotalAmount` is "totally" (pun intended) wrong.

Therefore there is no necessity to additionally check the count of Entities if you already checked `TotalAmount`.

From previos discussion, we know, that if `TotalAmount > 0 && Entities.Count() == 0` it's better to not return `NoContentResult`, because for paging case you loose important information that data (on other pages) exist.
That's why `IsNullOrEntitiesEmpty` shouldn't be ever used.

Reply:
Good evening, Vadym. thanks for your thoughts.

I am investigating usage of TotalAmount at the moment, and are going to continue doing this task tomorrow.

And, also, i decided to attract Dmytryi thoughts for our conversation, because, i am not sure, that we (you and I) are going in right direction, to finish this task. 

I can say, that i am a little bit confused at the moment, about TotalAmount naming and using.

So, good night, I will write to you tomorrow, after i push the new changes in branch. (also i will add Dmytriy thoughts about it, to new comment)

Reply:
Hello, Vadym.

As I promised:

You was right, your thoughts and Dmytryi explanations are match.

Dmytryi explanation:
The front end sends a request asking to show X records from the array with some filter, where X can be any number from 1 to 100 (as we agreed that this is our maximum).
We return X records and indicate that there are a total of Total records in the database that match this filter, where Total can be anything such as Total > X, Total = X, or Total < X.
Then the front end calculates how many pages with X records need to be displayed to show the Total records.
When the user clicks on a new page, the front end calculates how many records need to be skipped, like page_number * X, so that the back end knows how many records to skip for the new selection and can provide data from the required page.
Total amount is the total number of records matching the WHERE clause. Entity count is the number of records we retrieved from the database using the "take" parameter in the query.

So, my changes are:
- now we have extension method IsNullOrEmpty for SearchResult
- now we have extension method SearchResultToOkOrNoContent in ControllerExtensions
- fixed some tests (added TotalAmount to test, because, when TotalAmount = null, tests throws NullReferenceException)
- also, we have a bug checking in SearchResultToOkOrNoContent , when TotalAmount can be less than Entities.Count (Dmytriy said, that that case is real, and happend minimum one time)